### PR TITLE
some documentation improvements

### PR DESCRIPTION
**Pull request by p-alik**
no code changes, but make documentation examples newbie convenient

### CODE REVIEW

File: Data/Csv.hs [hunk 1]
```diff
@@ -221,6 +238,7 @@ import Data.Csv.Types
 -- parse a CSV file to a generic representation, just convert each
 -- record to a @'Vector' 'ByteString'@ value, like so:
 --
+-- > >>> import Data.ByteString
```
Comment (hvr) [inline Data/Csv.hs]: it's non-idiomatic to import Bytestring unqualified w/o an import list

File: Data/Csv.hs [hunk 2]
```diff
@@ -178,9 +190,13 @@ import Data.Csv.Types
 -- Derived:
 --
 -- > {-# LANGUAGE DeriveGeneric #-}
+-- > {-# LANGUAGE OverloadedStrings #-}
```
Comment (hvr) [inline Data/Csv.hs]: why is this needed here?

File: Data/Csv.hs [hunk 3]
```diff
@@ -140,24 +143,33 @@ import Data.Csv.Types
...
```
Comment (hvr) [inline Data/Csv.hs]: why is this needed here?

File: Data/Csv.hs [hunk 4]
```diff
@@ -140,24 +143,33 @@ import Data.Csv.Types
...
 -- > data Person = Person { name :: !Text , salary :: !Int }
--- >     deriving Generic
+-- >     deriving (Generic, Show)
 -- >
 -- > instance FromRecord Person
 -- > instance ToRecord Person
 --
 -- Manually defined:
 --
+-- > {-# LANGUAGE OverloadedStrings #-}
```
Comment (hvr) [inline Data/Csv.hs]: why is this needed here?

File: Data/Csv.hs [hunk 5]
```diff
@@ -116,6 +118,7 @@ import Data.Csv.Types
 --
 -- Decoding standard Haskell types:
 --
+-- > >>> import Data.Vector
```
Comment (hvr) [inline Data/Csv.hs]: please use import list

File: Data/Csv.hs (hunk 3 above)
Comment (p-alik) [inline Data/Csv.hs]: indeed, there is no need for `OverloadedStrings`. I'll remove it

File: Data/Csv.hs (hunk 4 above)
Comment (p-alik) [inline Data/Csv.hs]: `deriving Show` is for sake of `decode`. Without it `ghci` throws exception:
`No instance for (Show Person) arising from a use of ‘print’`

`OverloadedStrings` will be removed

File: Data/Csv.hs (hunk 2 above)
Comment (p-alik) [inline Data/Csv.hs]: `Show` for sake of `decode` in `ghci`.
`name'` and `age'` because `ghci` complains `This binding for ‘name’ shadows the existing binding`.
`OverloadedStrings` will be removed.

File: Data/Csv.hs (hunk 5 above)
Comment (p-alik) [inline Data/Csv.hs]: will be replaced by `import Data.Vector (Vector)`

File: Data/Csv.hs (hunk 1 above)
Comment (p-alik) [inline Data/Csv.hs]: will be replaced by `import Data.ByteString (ByteString)`

**Comment by hvr** [comment]
Thanks
